test/performance/odp_stash_perf.c

```diff
@@ -0,0 +1,501 @@
+
+	while (1) {
+		opt = getopt_long(argc, argv, shortopts, longopts, &long_index);
+
+		if (opt == -1)
+			break;
+
+		switch (opt) {
+		case 'c':
+			test_options->num_cpu = atoi(optarg);
+			break;
+		case 'n':
+			test_options->num_stash = atoi(optarg);
+			break;
+		case 'b':
+			test_options->max_burst = atoi(optarg);
+			break;
+		case 's':
+			test_options->stash_size = atoi(optarg);
+			break;
+		case 'r':
+			test_options->num_round = atoi(optarg);
+			break;
+		case 'm':
+			test_options->strict = 1;
+			break;
+		case 'h':
+			/* fall through */
+		default:
+			print_usage();
+			ret = -1;
+			break;
+		}
+	}
+
+	if (test_options->num_stash > MAX_STASHES) {
+		ODPH_ERR("Too many stashes %u. Test maximum %u.\n",
+			 test_options->num_stash, MAX_STASHES);
+		return -1;
+	}
+
+	return ret;
+}
+
+static int create_stashes(test_global_t *global)
+{
+	uint32_t i;
+	uint32_t tmp = 0;
+	test_options_t *test_options = &global->options;
+
+	uint32_t num_stash = test_options->num_stash;
+	uint32_t num_round = test_options->num_round;
+	odp_stash_t *stash = global->stash;
+	odp_stash_capability_t stash_capa;
+
+	printf("\nTesting %s stashes\n",
+	       test_options->strict == 0 ? "NORMAL" : "STRICT_SIZE");
+	printf("  num rounds           %u\n", num_round);
+	printf("  num stashes          %u\n", num_stash);
+	printf("  stash size           %u\n", test_options->stash_size);
+	printf("  max burst size       %u\n", test_options->max_burst);
+
+	if (odp_stash_capability(&stash_capa, ODP_STASH_TYPE_DEFAULT)) {
+		ODPH_ERR("Get stash capability failed\n");
+		return -1;
+	}
+
+	if (test_options->stash_size > stash_capa.max_num_obj) {
+		ODPH_ERR("Max stash size supported %" PRIu64 "\n",
+			 stash_capa.max_num_obj);
+		return -1;
+	}
+
+	if (test_options->num_stash > stash_capa.max_stashes) {
+		ODPH_ERR("Max stash supported %u\n", stash_capa.max_stashes);
+		return -1;
+	}
+
+	for (i = 0; i < num_stash; i++) {
+		odp_stash_param_t stash_param;
+
+		odp_stash_param_init(&stash_param);
+		stash_param.num_obj = test_options->stash_size;
+		stash_param.obj_size = sizeof(uint32_t);
+		stash_param.strict_size = test_options->strict;
+
+		stash[i] = odp_stash_create("test_stash_u32", &stash_param);
+		if (stash[i] == ODP_STASH_INVALID) {
+			ODPH_ERR("Stash create failed\n");
+			return -1;
+		}
+
+		int num_stored;
+		uint32_t num_remain = test_options->stash_size;
+
+		do {
+			num_stored = odp_stash_put_u32(stash[i], &tmp, num_remain);
+			if (num_stored < 0) {
+				ODPH_ERR("Error: Stash put failed\n");
+				return -1;
+			}
+			num_remain -= num_stored;
+		} while (num_stored == 0);
+	}
+
+	return 0;
+}
+
+static int destroy_stashes(test_global_t *global)
+{
+	odp_stash_t *stash = global->stash;
+	test_options_t *test_options = &global->options;
+	uint32_t num_stash = test_options->num_stash;
+	uint32_t tmp;
+	int num;
+
+	for (uint32_t i = 0; i < num_stash; i++) {
+		do {
+			num = odp_stash_get_u32(stash[i], &tmp,
+						test_options->max_burst);
```

Comment:
The output array for odp_stash_get_u32() should be of length `test_options->max_burst`.

Reply:
Updated with always get one object at a time.